back/app/services/nav_bar_item_service.rb

```diff
@@ -0,0 +1,56 @@
+class NavBarItemService
+  def auto_reposition!(item)
+    if !item.custom? && (position = candidate_position(item))
```

Comment:
It's intentional because I only want to calculate the candidate position once. I need to know if there exists a candidate position in the check (but this doesn't need to be calculated for custom items, which are always added to the end), and if there is a candidate position, move the item to that position.